src/parser_urdf_TEST.cc

```diff
@@ -994,6 +995,1096 @@ TEST(URDFParser, ParseWhitespace)
+    tinyxml2::XMLElement *model = sdf->FirstChildElement("model");
+    ASSERT_NE(nullptr, model);
+
+    tinyxml2::XMLElement *link = model->FirstChildElement("link");
+    ASSERT_NE(nullptr, link);
+    ASSERT_NE(nullptr, link->Attribute("name"));
+    EXPECT_EQ("link1", std::string(link->Attribute("name")));
+
+    // lumping occurs
+    tinyxml2::XMLElement *frame = model->FirstChildElement("frame");
+    ASSERT_NE(nullptr, frame);
+    ASSERT_NE(nullptr, frame->Attribute("name"));
+    EXPECT_EQ("joint1_2", std::string(frame->Attribute("name")));
+
+    frame = frame->NextSiblingElement("frame");
+    ASSERT_NE(nullptr, frame);
+    ASSERT_NE(nullptr, frame->Attribute("name"));
+    EXPECT_EQ("link2", std::string(frame->Attribute("name")));
+
+    link = link->NextSiblingElement("link");
+    ASSERT_NE(nullptr, link);
+    ASSERT_NE(nullptr, link->Attribute("name"));
+    EXPECT_EQ("link3", std::string(link->Attribute("name")));
+
+    tinyxml2::XMLElement *joint = model->FirstChildElement("joint");
+    ASSERT_NE(nullptr, joint);
+    ASSERT_NE(nullptr, joint->Attribute("name"));
+    EXPECT_EQ("joint2_3", std::string(joint->Attribute("name")));
+  }
+
+  // intermediate link with parent fixed joint, with lumping turned off, don't
+  // convert into revolute joints, preserve fixed joints
+  {
+    // clear the contents of the buffer
+    buffer.str("");
+
+    std::string str = R"(
+      <robot name='test_robot'>
+        <link name='link1'>
+          <inertial>
+            <mass value='0.1' />
+            <origin rpy='1.570796326794895 0 0' xyz='0.123456789123456 0 0.0' />
+            <inertia ixx='0.01' ixy='0' ixz='0' iyy='0.01' iyz='0' izz='0.01' />
+          </inertial>
+        </link>
+        <link name='link2'/>
+        <joint name='joint1_2' type='fixed'>
+          <parent link='link1' />
+          <child  link='link2' />
+          <origin xyz='1 2 3' rpy='0.0 0.0 1.57'/>
+        </joint>
+        <link name='link3'>
+          <inertial>
+            <mass value='0.1' />
+            <origin rpy='1.570796326794895 0 0' xyz='0.123456789123456 0 0.0' />
+            <inertia ixx='0.01' ixy='0' ixz='0' iyy='0.01' iyz='0' izz='0.01' />
+          </inertial>
+        </link>
+        <joint name='joint2_3' type='continuous'>
+          <parent link='link2' />
+          <child  link='link3' />
+          <origin xyz='0.0 0.0 0.0' rpy='0.0 0.0 1.57'/>
+        </joint>
+        <gazebo reference='joint1_2'>
+          <disableFixedJointLumping>true</disableFixedJointLumping>
+        </gazebo>
+        <gazebo reference='joint1_2'>
+          <preserveFixedJoint>true</preserveFixedJoint>
+        </gazebo>
+      </robot>)";
+
+    sdf::URDF2SDF parser;
+    sdf::ParserConfig config;
+    config.URDFSetPreserveFixedJoint(true);
+    tinyxml2::XMLDocument sdfResult;
+    parser.InitModelString(str, config, &sdfResult);
+
+    // link2 will be converted to a frame, relative to link1, joint1_2 will be
```

Comment:
I realized it might not be a good idea to completely remove `joint1_2`. Can we convert to a frame such that `joint1_2` is attached to `link2` and `link2` is attached to `link1`?

Reply:
oh, that is such a better idea. I went ahead to remove the joint since the cases URDF allows are simpler, and there is no way for a previous joint to be referenced, except for when transmissions, sensors or gazebo tags are involved.

Just added it back, and have it attached to the converted link, https://github.com/gazebosim/sdformat/pull/1238/commits/a575e1c852e6a0077eba24e5b07a11ae613f8e2e